### ISSUE TITLE
Od causes compilation failure due to inability to resolve immediate values.

### ISSUE THREAD

**Issue by adam-yang** [Microsoft/DirectXShaderCompiler#856]
While compiling the following code:
```
// od_bug.hlsl
Texture2D tex : register(t0);
sampler smpl : register(s0);

float4 DoSample(float2 pos, int2 off) {
	return tex.Sample(smpl, pos.xy, off.xy);
}

float4 main(float2 uv : TEX_COORD) : SV_Target {
	float4 center = DoSample(uv, int2(0, 0));
	float4 left = DoSample(uv, int2(-1, 0));
	float4 right = DoSample(uv, int2(1, 0));
	return (center + left + right) / 3.0;
}
```

with the following command:
```
dxc od_bug.hlsl /T ps_5_0 /Od
```

Gives the following error:
```
error:  Offsets for Sample* must be immediated value. Consider unroll the loop manually and use O3, it may help in some cases Offsets for Sample* must be immediated value. Consider unroll the loop manually and use O3, it may help in some cases Offsets for Sample* must be immediated value. Consider unroll the loop manually and use O3, it may help in some cases Offsets for Sample* must be immediated value. Consider unroll the loop manually and use O3, it may help in some cases Offsets for Sample* must be immediated value. Consider unroll the loop manually and use O3, it may help in some cases Offsets for Sample* must be immediated value. Consider unroll the loop manually and use O3, it may help in some cases
```

**Comment by vcsharma** [comment]
This should have been resolved by #1417 ?

**Comment by MrTrillian** [comment]
Verified not repro with newer Windows SDKs and on master.

```
>"C:\Program Files (x86)\Windows Kits\10\bin\10.0.17134.0\x64\dxc.exe" /Od /T ps_5_0 od_bug.hlsl
error:  Offsets for Sample* must be immediated value. Consider unroll the loop manually and use O3, it may help in some cases Offsets

>"C:\Program Files (x86)\Windows Kits\10\bin\10.0.17763.0\x64\dxc.exe" /Od /T ps_5_0 od_bug.hlsl
;
; Input signature:
;
; Name                 Index   Mask Register SysValue  Format   Used
; -------------------- ----- ------ -------- -------- ------- ------
; TEX_COORD                0   xy          0     NONE   float
;
[... - no error]
```
